### PR TITLE
FreeBSD doesn't have SIGPOLL

### PR DESCRIPTION
**Pull request by rbgarga**

### CODE REVIEW

File: luaa.c [hunk 1]
```diff
@@ -612,7 +613,9 @@ setup_awesome_signals(lua_State *L)
 
     /* POSIX.1-2001, according to man 7 signal */
     SETUP_SIGNAL(SIGBUS);
+#ifndef __FreeBSD__
     SETUP_SIGNAL(SIGPOLL);
+#endif
```
Comment (blueyed) [inline luaa.c]: Shouldn't this check for if `SIGPOLL` is defined?!  (and then leave a comment about FreeBSD there only)

Comment (rbgarga) [inline luaa.c]: Yeah, it should be fine and we will avoid extra include. Let me change it

**Comment by psychon** [comment]
Sigh. And I even checked the man page... Does this mean that FreeBSD does not follow SUSv2 / POSIX.1-2001?

Anyway, thanks for fixing this! I hope this is the worst problem that we bring to you and building onFreeBSD.

**Comment by blueyed** [comment]
Thanks!

**Comment by rbgarga** [comment]
@psychon I didn't follow all the history, but there is an old discussion here:
https://lists.freebsd.org/pipermail/freebsd-standards/2011-January/002098.html

Thanks!

**Pull request merged**